### PR TITLE
Added doxyfile in rosidl_runtime_c and rosidl_runtime_cpp

### PR DESCRIPTION
**Pull request by ahcorde**
Added Doxyfile to `rosidl_runtime_c` and `rosidl_runtime_cpp` in the process of reaching Quality Level 1.

Signed-off-by: ahcorde <ahcorde@gmail.com>

### CODE REVIEW

File: rosidl_runtime_c/Doxyfile [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+GENERATE_LATEX         = NO
+
+ENABLE_PREPROCESSING   = YES
+MACRO_EXPANSION        = YES
+EXPAND_ONLY_PREDEF     = YES
+PREDEFINED             += ROSIDL_GENERATOR_C_PUBLIC=
+
+# Tag files that do not exist will produce a warning and cross-project linking will not work.
+TAGFILES += "../../../../doxygen_tag_files/cppreference-doxygen-web.tag.xml=http://en.cppreference.com/w/"
+# Consider changing "latest" to the version you want to reference (e.g. beta1 or 1.0.0)
```
Comment (brawner) [inline rosidl_runtime_c/Doxyfile:23]: Remove

File: rosidl_runtime_cpp/Doxyfile [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+GENERATE_LATEX         = NO
+
+ENABLE_PREPROCESSING   = YES
+MACRO_EXPANSION        = YES
+EXPAND_ONLY_PREDEF     = YES
+
+# Tag files that do not exist will produce a warning and cross-project linking will not work.
+TAGFILES += "../../../../doxygen_tag_files/cppreference-doxygen-web.tag.xml=http://en.cppreference.com/w/"
+# Consider changing "latest" to the version you want to reference (e.g. beta1 or 1.0.0)
```
Comment (brawner) [inline rosidl_runtime_cpp/Doxyfile:22]: Remove

File: rosidl_runtime_c/Doxyfile [hunk 3]
```diff
@@ -0,0 +1,26 @@
+# All settings not listed here will use the Doxygen default values.
+
+PROJECT_NAME           = "rosidl_runtime_c"
+PROJECT_NUMBER         = master
+PROJECT_BRIEF          = "Generate the ROS interfaces in C."
```
Comment (brawner) [inline rosidl_runtime_c/Doxyfile:5]:
```suggestion
PROJECT_BRIEF          = "Generate the rosidl interfaces in C."
```

File: rosidl_runtime_cpp/Doxyfile [hunk 4]
```diff
@@ -0,0 +1,25 @@
+# All settings not listed here will use the Doxygen default values.
+
+PROJECT_NAME           = "rosidl_runtime_cpp"
+PROJECT_NUMBER         = master
+PROJECT_BRIEF          = "Generate the ROS interfaces in C++."
```
Comment (brawner) [inline rosidl_runtime_cpp/Doxyfile:5]:
```suggestion
PROJECT_BRIEF          = "Generate the rosidl interfaces in C++."
```

Comment (ahcorde) [reply rosidl_runtime_cpp/Doxyfile]: Done

**Pull request merged**